amethyst_audio/src/systems/audio.rs

```diff
@@ -62,23 +61,20 @@ impl<'a> System<'a> for AudioSystem {
                 .and_then(|sl| transform.get(sl.0))
                 .or_else(|| transform.get(entity))
             {
-                let listener_transform = listener_transform.0;
-                let left_ear_position =
-                    listener_transform.transform_point(listener.left_ear).into();
-                let right_ear_position = listener_transform
-                    .transform_point(listener.right_ear)
-                    .into();
+                let listener_transform = listener_transform.0.column(3).xyz();
+                let left_ear_position = listener_transform + listener.left_ear;
+                let right_ear_position = listener_transform + listener.right_ear;
                 for (transform, mut audio_emitter) in (&transform, &mut audio_emitter).join() {
-                    let x = transform.0[3][0];
-                    let y = transform.0[3][1];
-                    let z = transform.0[3][2];
+                    let x = transform.0[(0, 3)];
+                    let y = transform.0[(1, 3)];
+                    let z = transform.0[(2, 3)];
```

Comment:
This does look like it's reversed, but I guess @sebcrozet can say if it actually is or not. :stuck_out_tongue:

Reply:
It is not reversed. For example `transform.0[3][0]` returns the element at column 3 and row 0 because the first indexing with a single index returns a column of the matrix. The `transform.0[(0, 3)]` is also the element at column 3 and row 0 because it follows the natural order of indices in mathematics (row first, column second).